phylanx/execution_tree/primitives/sub_operation.hpp

```diff
@@ -0,0 +1,50 @@
+//  Copyright (c) 2017  Bibek Wagle
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(PHYLANX_PRIMITIVES_SUB_OPERATION_SEP_15_2017_1035AM)
+#define PHYLANX_PRIMITIVES_SUB_OPERATION_SEP_15_2017_1035AM
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/execution_tree/primitives/base_primitive.hpp>
+
+#include <hpx/include/components.hpp>
+
+#include <utility>
+#include <vector>
+
+namespace phylanx { namespace execution_tree { namespace primitives
+        {
+            class HPX_COMPONENT_EXPORT sub_operation
+                    : public base_primitive
+                            , public hpx::components::component_base<sub_operation>
+            {
+            private:
+                using operands_type = std::vector<ir::node_data<double>>;
+
+            public:
+                sub_operation() = default;
+
+                sub_operation(std::vector<primitive> && operands);
+                sub_operation(std::vector<primitive> const& operands);
+
+                hpx::future<ir::node_data<double>> eval() const override;
+
+            protected:
+                ir::node_data<double> sub0d(operands_type const& ops) const;
+                ir::node_data<double> sub1d(operands_type const& ops) const;
+                ir::node_data<double> sub2d(operands_type const& ops) const;
+
+                ir::node_data<double> sub1d1d(operands_type const& ops) const;
+                ir::node_data<double> sub2d2d(operands_type const& ops) const;
+
+            private:
+                std::vector<primitive> operands_;
+            };
+        }}}
+
```

Comment:
What happened to the indentation here?

Reply:
It was CLion wanting to change formatting automatically :( This is fixed with db121c1c1bcb974caf19314a2865ec5ec6038dc0